### ISSUE TITLE
Implement React hooks 16.12.0

### ISSUE THREAD

**Issue by thisisamardeep** [desktop/desktop#8850]
### Describe the feature or problem you’d like to solve

Right now we are on React 16.3.2 .I have created a draft implementation ,removed state and used Hooks in the entire Project.There are some issues which i am working on.
It should be ready by 3rd week of Jan.

A clear and concise description of what the feature or problem is.

### Proposed solution

How will it benefit Desktop and its users?

### Additional context

Add any other context like screenshots or mockups are helpful, if applicable.

**Comment by niik** [comment]
Hi @thisisamardeep, thanks for wanting to contribute!

Before you spend too much time on this I want to set some expectations and try to explain our thinking when it comes to upgrading our dependencies.

A good place to start is our documentation for [upgrading dependencies.](https://github.com/desktop/desktop/blob/development/docs/contributing/upgrading-dependencies.md).

For something like upgrading React we'd like to start first by looking at the benefits and risks of doing so. We'd be very interested in hearing about your reasons for wanting to upgrade as well as anything interesting you've learned from the work you've already completed before we make a decision on upgrading.

There's inherent risk in any change and we seek to minimize that risk. As such I propose we start by identifying if there's enough value in us upgrading to React 16.12 on its own, without any (or minimal) other changes to our logic (such as using hooks). Examples of things that would encourage us to do so would be perceivable performance, security, and/or stability improvements. Given that [our policy](https://github.com/desktop/desktop/blob/development/docs/contributing/upgrading-dependencies.md) is to stay at least one version behind the latest React release we could start the upgrade process now and aim to be ready when 16.13 is released.

After we've landed the upgrade itself we can start discussing the benefits of us leaning more heavily on hooks over the conventional approach we have today. I'd again like to hear your thoughts on why you think a hook-based approach would be beneficial for the Desktop project. If we decide that it's the right way to go I'd still want us to do a gradual approach where we convert a set of components over a number of releases to minimize the risk of users experiencing issues.

To sum things up, let's have a conversation about upgrading first in the hope of avoiding unnecessary work.

**Comment by niik** [comment]
Thank you for your issue!

We haven't gotten a response to our questions.
With only the information that is currently in the issue, we don't have enough
information to take action. I'm going to close this but don't hesitate to reach
out if you have or find the answers we need, we'll be happy to reopen the
issue.
